### ISSUE TITLE
tf.signal is differentiable

### ISSUE THREAD

**Issue by rryan** [KinWaiCheuk/nnAudio#22]
Hi there,

I'm the author of Tensorflow's `tf.signal` package. Your paper says that `tf.signal` does not support gradients, however this is not true. All operations in `tf.signal` are fully differentiable and come with GPU and TPU support. Could you please update your paper on arXiv to correct this?

**Comment by rryan** [comment]
I think you were trying to say that it does not support a trainable basis? That's true, but it's still differentiable so you can incorporate it into a network and backpropagate through it (e.g. to add a spectral loss to model that generates time-domain waveforms -- as in [Parallel WaveNet](https://arxiv.org/abs/1711.10433))

**Comment by rryan** [comment]
Nice job on this repo and the paper by the way :).

**Comment by KinWaiCheuk** [comment]
> I think you were trying to say that it does not support a trainable basis? That's true, but it's still differentiable so you can incorporate it into a network and backpropagate through it (e.g. to add a spectral loss to model that generates time-domain waveforms -- as in [Parallel WaveNet](https://arxiv.org/abs/1711.10433))

Thanks for pointing it out. You are right, I was talking about the trainable basis. I will fix the paper now.

**Comment by KinWaiCheuk** [comment]
Hi Ryan, I have removed the claim that `tf.signal` doesn't support backpropagation. 
Can you check if it is better now? 
https://arxiv.org/abs/1912.12055

**Comment by rryan** [comment]
Thanks a lot for updating it!

**Comment by antonwnk** [comment]
Hi. Can I ask what is meant by trainable basis?
I came across this issue while googling around to see whether I could optimize `tf.signal.stft`'s `frame_length` and `frame_step` parameters along with all the other parameters of my network, through backprop. Would this be technically possible?
